lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -596,6 +615,18 @@ translate_remote(maps, merge, Meta, [Map1, Map2], S) ->
     {[TMap1, TMap2], TS} ->
       {{call, Ann, {remote, Ann, {atom, Ann, maps}, {atom, Ann, merge}}, [TMap1, TMap2]}, TS}
   end;
+translate_remote(Left, Right, Meta, [], S)
+    when (Left =:= nil orelse is_boolean(Left)), is_atom(Right) ->
+  Ann = ?ann(Meta),
+  TLeft = {atom, Ann, Left},
+  TRight = {atom, Ann, Right},
+  case proplists:get_value(no_parens, Meta, false) of
+    true ->
+      TError = {tuple, Ann, [{atom, Ann, badkey}, TRight, TLeft]},
+      {?remote(Ann, erlang, error, [TError]), S};
+    false ->
+      {{call, Ann, {remote, Ann, TLeft, TRight}, []}, S}
+  end;
```

Comment:
I think we don't need this, correct? we can just emit the original code and let it fail as usual?

Reply:
This is the case where `nil.id` is being called (as opposed to `x = nil; x.id`).

Without this, we would get:
<img width="1158" alt="Screen Shot 2022-07-26 at 18 20 11" src="https://user-images.githubusercontent.com/11598866/180971967-b75c05ca-0c8a-4695-aed0-3a670982b363.png">

But maybe there's a better way: we can replace the `translate` clause above to check for these earlier: https://github.com/elixir-lang/elixir/pull/12003/commits/55a3e64b8afb8bc2f075cda9e45a4c53ecc44664

WDYT?

Reply:
The tradeoff is that we generate a bit of a garbage `case` for `nil.id`, but this should not really be a thing for production code anyway, so it probably doesn't have to be optimal It just have to be correct when you try it out in IEx?